target/bin/debug-getmail

```diff
@@ -5,9 +5,17 @@ source /usr/local/bin/helpers/log.sh
 # shellcheck source=../scripts/startup/setup-stack.sh
 source /usr/local/bin/setup.d/getmail.sh
 
-_setup_getmail
+# Setup getmail, even if not enabled.
+ENABLE_GETMAIL=1 _setup_getmail
```

Comment:
I see no downside in having the ability to debug, even when the service is not enabled on container start.

Reply:
Yeah... but in the unlikely event the user did think to add the ENV var to toggle the feature in the first place, these two debug commands would not catch that.

Typically we'd expect them to run the debug command from the same container config where they'd have ENV configured, so `docker compose run dms setup debug ...` should use that ENV, or equivalent for `docker run`. The only time it would not is if they ignored their existing config and were not ok with providing that ENV despite that?

It's an unlikely scenario, so I'm fine if you want to ignore that. Perhaps it's an issue if you use the `setup.sh` wrapper as a convenience shortcut to the `docker` commands we advise for running the `setup` CLI (_these are typically `docker exec` I think, so ignore expectation of providing additional config, but `setup dkim` may expect it_).

Reply:
> Yeah... but in the unlikely event the user did think to add the ENV var to toggle the feature in the first place, these two debug commands would not catch that.

`debug-getmail.sh` / `getmail --dump` does no error checking. It only dumps the processed configuration, i.e. everything that got configured explicitly and implicit (default values). Wrong values btw are ignored and removed from the output.

---

I don't understand the issue you are trying to explain :-\ Using `ENABLE_FETCHMAIL=1 _setup_fetchmail` adds an additional possibility: to use the debug command (when getmail is disabled).

One might it call it an "logic error" to allow it, some else may call it a feature 🤷 

I can't think of a real disadvantage due to this. Some might want this feature, to check/verify the configuration before enabling the service. 

---

Like other setup functions, the ENV check could be moved to `start-mailserver.sh`:
https://github.com/docker-mailserver/docker-mailserver/blob/fb57905aa3248ee215fc96aad16ec203a8014975/target/scripts/start-mailserver.sh#L107-L127

That would remove the necessity to call `_setup_fetchmail` with `ENABLE_FETCHMAIL=1` prefixed.

---

If you are fine with it, I'd like to leave it as it is for now. The debug-(getmail|fetchmail) commands can be improved in a separate PR.